packages/bff/src/services/catalogService.ts

```diff
@@ -294,5 +306,79 @@ export function catalogServiceBuilder(
         id: updatedEservice.id,
       };
     },
+    getProducerEServices: async (
+      eserviceName: string | undefined,
+      consumersIds: string[],
+      offset: number,
+      limit: number,
+      { headers, authData }: WithLogger<BffAppContext>
+    ): Promise<bffApi.ProducerEServices> => {
+      const producerId = authData.organizationId;
+      const response: {
+        results: catalogApi.EService[];
+        totalCount: number;
+      } = {
+        results: [],
+        totalCount: 0,
+      };
+
+      if (consumersIds.length === 0) {
+        const { results, totalCount } = await catalogProcessClient.getEServices(
+          {
+            headers,
+            queries: {
+              name: eserviceName,
+              producersIds: producerId,
+              offset,
+              limit,
+            },
+          }
+        );
+
+        response.results = results;
+        response.totalCount = totalCount;
+      } else {
+        const eserviceIds = (
+          await getAllFromPaginated(async (offset: number, limit: number) =>
+            agreementProcessClient.getAgreements({
+              headers,
+              queries: {
+                consumersIds,
+                producersIds: [producerId],
+                eservicesIds: [],
+                states: [],
+                offset,
+                limit,
+              },
+            })
+          )
+        ).map((agreement) => agreement.eserviceId);
+
+        const { results, totalCount } = await catalogProcessClient.getEServices(
```

Comment:
We should call this only if `eserviceIds.length !== 0`; otherwise `response` should just be empty